design/device-set.md

```diff
@@ -0,0 +1,50 @@
+# Rook Device Sets
+
+**Target version**: Rook 1.1
+
+## Background
+The primary motivation for this feature is to take advantage of the mobility of storage in cloud-based environments by defining storage based on sets of devices consumed as PVCs. In environments like AWS you can request remote storage that can be dynamically provisioned and attached to any node in a given Availability Zone (AZ). The design can also accommodate non-cloud environments through the use of local PVs.
+
+## DeviceSet struct
+```go
+struct DeviceSet {
```

Comment:
It seems to me that the purpose of `DeviceSet`s is to allow Rook to use storage classes as backing storage. I don't see any use for this beyond consumption of storage classes for devices. To this end, I think DeviceSet isn't the best name for this. It isn't very specific, and it seems like it would be a lot clearer to users if we name this something like `StorageClassDeviceSet`.

Reply:
Not a bad idea.